package/sbp_anpp_bridge/src/sbp_anpp_bridge.c

```diff
@@ -83,44 +97,45 @@ static int notify_time_offset_changed(void *context)
   return 0;
 }
 
-static void convert_anpp_odometer_to_sbp(odometer_packet_t *an_odo_packet, msg_odometry_t *odo_msg)
+static void convert_anpp_odometer_to_sbp(odometer_packet_t *an_odo_packet,
+                                         msg_odometry_t *odo_msg,
+                                         anpp_bridge_ctx_t *context)
 {
-  if (last_gpstime_tow != INVALID_TOW) {
-    double tow = last_gpstime_tow;
-    struct timespec now;
-    clock_gettime(CLOCK_MONOTONIC, &now);
-    /* Add on elapsed time since gps_time_callback was received */
-    tow += (now.tv_sec * 1000.0 + now.tv_nsec / 1000000.0)
-           - (last_gps_systime.tv_sec * 1000.0 + last_gps_systime.tv_nsec / 1000000.0);
-    tow += odo_time_offset_ms;
-    if (tow > MS_IN_WEEK) {
-      tow -= MS_IN_WEEK;
-    }
-    if (tow < 0) {
-      tow += MS_IN_WEEK;
-    }
-    odo_msg->tow = (u32)tow;
-    odo_msg->velocity = (s32)M_TO_MM(an_odo_packet->speed);
-    odo_msg->flags = (u8)(SBP_MSG_ODOMETRY_FLAGS_SRC0 | SBP_MSG_ODOMETRY_FLAGS_SYSTIME);
+  double tow = context->last_gpstime_tow;
+  struct timespec now;
+  clock_gettime(CLOCK_MONOTONIC, &now);
```

Comment:
maybe we can foist this helper from `manage_led.c` into `libpiksi/util`
https://github.com/swift-nav/piksi_buildroot/blob/e01649bcf3cf27b69cf67fa3f070ca831bab8bfc/package/piksi_leds/src/manage_led.c#L103